### PR TITLE
Fix start and end date functionality and add tests

### PR DESCRIPTION
**Pull request by AballahNsour**
# *Name of the PR*

## :recycle: Current situation & Problem
*Link any open issues or pull requests (PRs) related to this PR. Please ensure that all non-trivial PRs are first tracked and discussed in an existing GitHub issue or discussion.*

## :gear: Release Notes 
*Add a bullet point list summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented or links to the documentation if it appends or changes the public interface.*

## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 

## :white_check_mark: Testing
*Please ensure that the PR meets the testing requirements set by CodeCov and that new functionality is appropriately tested.*
*This section describes important information about the tests and why some elements might not be testable.*

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PSchmiedmayer** [comment]
@AballahNsour Thank you for taking a look at the issue! Feel free to create a PR and tag us if you have any questions. Always great to see contributions from outside contributors and we are happy to support you in the process!

**Comment by AballahNsour** [comment]
forgive me I went though the rules again it happens that i have vaiolated one of contribution rules i Will revert on the changes and I will start over.

**Comment by PSchmiedmayer** [comment]
No worries; it is always a challenge to get started.

Let us know how we can be helpful, we always appreciate contributions and improvements 🚀

**Comment by AballahNsour** [comment]
@PSchmiedmayer hello, I did the best i could with adding the two test cases. 
I may need more assistance, but will learn fast :)

File: Tests/SpeziSchedulerTests/SchedulerTests.swift [hunk 1]
```diff
@@ -286,18 +286,49 @@ final class SchedulerTests: XCTestCase {
...
-        
+
         var completedEventHasher = Hasher()
         completedEvent.hash(into: &completedEventHasher)
         let completedEventHash = completedEventHasher.finalize()
-        
+
         XCTAssertEqual(scheduledEventHash, completedEventHash)
     }
+    func testTaskSchedulingWithStartDateAndEndDate() async throws {
```
Comment (PSchmiedmayer) [inline Tests/SpeziSchedulerTests/SchedulerTests.swift:303]: Please add empty lines between different functions:
```suggestion
    
    func testTaskSchedulingWithStartDateAndEndDate() async throws {
```

Comment (PSchmiedmayer) [inline Tests/SpeziSchedulerTests/SchedulerTests.swift:303]: Thanks for adding this test!

File: Tests/SpeziSchedulerTests/SchedulerTests.swift [hunk 2]
```diff
@@ -286,18 +286,49 @@ final class SchedulerTests: XCTestCase {
...
+        let startDate = Date().addingTimeInterval(3_000)
```
Comment (PSchmiedmayer) [inline Tests/SpeziSchedulerTests/SchedulerTests.swift:304]: A general suggestion for writing the tests: I would avoid using `.now` (or `Date()`) for dates. We had some issues with this here and in other packages. I would rather load dates based on fixed date and time combinations and use that for reasoning in the tests. Makes testing and communicating about the tests + asserting correct responses way easier.

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
@AballahNsour thank you for working on the PR and adding some additional tests to this package; greatly appreciated!

This PR is a good step in the right direction; we are still missing some context from #21 that would allow us to close the issue.

1. We successfully test that a task that has an end date works. I would suggest that we schedule a tasks that hast at least two scheduled times between the start and end date.
2. We need to test the [`events(from,to)`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/0.8.2/documentation/spezischeduler/task/events(from:to:)) function in a separate unit test (to fully address #21). This should include at task that has a fixed end date and one that might have a number of events and we should then retrieve taks up to the end date (and expect all the them to get back) and requests tasks up to an end data that should only return parts of the tasks.

A general suggestion for writing the tests: I would avoid using `.now` (or `Date()`) for dates. We had some issues with this here and in other packages. I would rather load dates based on fixed date and time combinations and use that for reasoning in the tests. Makes testing and communicating about the tests + asserting correct responses way easier.

**Pull request closed without merging**